lib/blacklight/access_controls/enforcement.rb

```diff
@@ -23,6 +23,14 @@ def current_ability
         @current_ability || raise("current_ability has not been set on #{self}")
       end
 
+      def except(*)
+        super.tap { |new_builder| new_builder.current_ability = current_ability }
+      end
+
+      def append(*)
+        super.tap { |new_builder| new_builder.current_ability = current_ability }
+      end
+
```

Comment:
@awead the `super` keyword calls the current method in the super class. On line 27 that would be `append` (https://github.com/projectblacklight/blacklight/blob/master/lib/blacklight/search_builder.rb#L54). On line 31 it calls `except` (https://github.com/projectblacklight/blacklight/blob/master/lib/blacklight/search_builder.rb#L76)